src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,555 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
+import "../interfaces/ISocketUpdater.sol";
+import "../interfaces/IServiceManager.sol";
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IVoteWeigher.sol";
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IPauserRegistry.sol";
+
+import "../libraries/EIP1271SignatureUtils.sol";
+import "../libraries/BitmapUtils.sol";
+import "../libraries/MiddlewareUtils.sol";
+
+import "../permissions/Pausable.sol";
+
+/**
+ * @title A `RegistryCoordinator` that has three registries:
+ *      1) a `StakeRegistry` that keeps track of operators' stakes
+ *      2) a `BLSPubkeyRegistry` that keeps track of operators' BLS public keys and aggregate BLS public keys for each quorum
+ *      3) an `IndexRegistry` that keeps track of an ordered list of operators for each quorum
+ * 
+ * @author Layr Labs, Inc.
+ */
+contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistryCoordinatorWithIndices, ISocketUpdater, Pausable {
+    using BN254 for BN254.G1Point;
+
+    /// @notice The EIP-712 typehash for the `DelegationApproval` struct used by the contract
+    bytes32 public constant OPERATOR_CHURN_APPROVAL_TYPEHASH =
+        keccak256("OperatorChurnApproval(bytes32 registeringOperatorId, OperatorKickParam[] operatorKickParams)OperatorKickParam(address operator, BN254.G1Point pubkey, bytes32[] operatorIdsToSwap)BN254.G1Point(uint256 x, uint256 y)");
+
+    uint16 internal constant BIPS_DENOMINATOR = 10000;
+
+    /// @notice Index for flag that pauses operator registration
+    uint8 internal constant PAUSED_REGISTER_OPERATOR = 0;
+    /// @notice Index for flag that pauses operator deregistration
+    uint8 internal constant PAUSED_DEREGISTER_OPERATOR = 1;
+
+    /// @notice the EigenLayer Slasher
+    ISlasher public immutable slasher;
+    /// @notice the Service Manager for the service that this contract is coordinating
+    IServiceManager public immutable serviceManager;
+    /// @notice the BLS Pubkey Registry contract that will keep track of operators' BLS public keys
+    IBLSPubkeyRegistry public immutable blsPubkeyRegistry;
+    /// @notice the Stake Registry contract that will keep track of operators' stakes
+    IStakeRegistry public immutable stakeRegistry;
+    /// @notice the Index Registry contract that will keep track of operators' indexes
+    IIndexRegistry public immutable indexRegistry;
+
+    /// @notice the mapping from quorum number to the maximum number of operators that can be registered for that quorum
```

Comment:
"maximum number of operators that can be registered for that quorum" -- there's more info stored in this struct, right?